pkg/util/tests/kubectl/kubectl.go

```diff
@@ -0,0 +1,335 @@
+	binary := DefaultBinary
+	if cmd := os.Getenv("COMMAND"); cmd != "" {
+		binary = cmd
+	}
+
+	return binary + " " + fmt.Sprintf(format, args...)
+}
+
+// ApplyString applies the given yaml string to the cluster
+func ApplyString(yamlString string) error {
+	cmd := kubectl("apply --server-side -f -")
+	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		return fmt.Errorf("error applying yaml: %v", err)
+	}
+
+	return nil
+}
+
+// DeleteString delete the given yaml string to the cluster
+func DeleteString(yamlString string) error {
+	cmd := kubectl("delete -f -")
+	output, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		// Workaround because the resource can be already deleted
+		if strings.Contains(output, "not found") {
+			return nil
+		}
+
+		return fmt.Errorf("error deleting yaml: %v", err)
+	}
+
+	return nil
+}
+
+// GetConditions returns the condition of a resource
+func GetConditions(ns, kind, name string) ([]r.Condition, error) {
+	output, err := GetJSON(ns, kind, name)
+	if err != nil {
+		return []r.Condition{}, err
+	}
+
+	var resource r.Resource
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return []r.Condition{}, err
+	}
+
+	return resource.Status.Conditions, nil
+}
+
+// GetPodPhase returns the phase of a pod
+func GetPodPhase(ns, selector string) (string, error) {
+	podName, err := GetPodName(ns, selector)
+	if err != nil {
+		return "", err
+	}
+
+	output, err := GetJSON(ns, "pod", podName)
+	if err != nil {
+		return "", err
+	}
+
+	var resource r.Resource
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return "", err
+	}
+
+	return resource.Status.Phase, nil
+}
+
+// GetCRDs returns all the CRDs names in a list
+func GetCRDs() ([]string, error) {
+	cmd := kubectl("get crds -o name")
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return []string{}, fmt.Errorf("error getting crds: %v", err)
+	}
+	return extractNames(output), nil
+}
+
+// GetResourceList returns a json list of the resources of a namespace by resource name
+func GetResourceList(ns, kind string) (r.ResourceList, error) {
+	// TODO: improve the function to get all the resources
+	output, err := GetJSON(ns, kind, "")
+	if err != nil {
+		return EmptyResourceList, err
+	}
+
+	var resourceList r.ResourceList
+	err = json.Unmarshal([]byte(output), &resourceList)
+	if err != nil {
+		return EmptyResourceList, err
+	}
+
+	// Return an empty list if there are no resources
+	if len(resourceList.Items) == 0 {
+		return EmptyResourceList, nil
+	}
+
+	return resourceList, nil
+}
+
+// GetJSON returns the json of a resource
+// Arguments:
+// - ns: namespace
+// - kind: type of the resource
+// - name: name of the resource
+func GetJSON(ns, kind, name string) (string, error) {
+	nsflag := "-n"
+	if ns == "" {
+		nsflag = "--all-namespaces"
+	}
+	cmd := kubectl("get %s %s %s %s -o json", kind, name, nsflag, ns)
```

Comment:
```suggestion
	nsflag := "-n " + ns
	if ns == "" {
		nsflag = "--all-namespaces"
	}
	cmd := kubectl("get %s %s %s -o json", kind, name, nsflag)
```

Or better yet, introduce function `nsFlag(ns string)` and call it directly: `kubectl("get ...", kind, name, nsFlag(ns))`